src/clj/fluree/db/query/exec/eval.cljc

```diff
@@ -326,6 +316,40 @@
     (or (= a b)
         (pos? (compare* a a-dt b b-dt)))))
 
+(defn max
+  [coll]
+  (let [compare-fn (fn [a b]
+                     (if (pos? (compare* (:value a) (:datatype-iri a)
+                                         (:value b) (:datatype-iri b)))
+                       a
+                       b))
+        ;; I think we should also return the datatype-iri here, 
+        ;; but currently for date-like values, that is java.time.LocalDate 
+        ;; or java.time.OffsetDateTime, which ought actually to be returned 
+        ;; as xsd:date or xsd:dateTime, respectively.
+        ;;
+        ;;     {:keys [value datatype-iri]} (reduce compare-fn coll)]
+        ;; (where/->typed-val value datatype-iri)))
+        value      (:value (reduce compare-fn coll))]
+    (where/->typed-val value)))
+
+(defn min
+  [coll]
+  (let [compare-fn (fn [a b]
+                     (if (neg? (compare* (:value a) (:datatype-iri a)
+                                         (:value b) (:datatype-iri b)))
+                       a
+                       b))
+        ;; I think we should also return the datatype-iri here, 
+        ;; but currently for date-like values, that is java.time.LocalDate 
+        ;; or java.time.OffsetDateTime, which ought actually to be returned 
+        ;; as xsd:date or xsd:dateTime, respectively.
+        ;;
+        ;;     {:keys [value datatype-iri]} (reduce compare-fn coll)]
+        ;; (where/->typed-val value datatype-iri)))
+        value      (:value (reduce compare-fn coll))]
```

Comment:
I think we could just return the value produced by `reduce`, which will be a TypedValue with the correct datatype (and lang, if applicable).